L1Trigger/L1THGCal/plugins/backend/HGCalTowerProcessor.cc

```diff
@@ -33,33 +34,39 @@ class HGCalTowerProcessor : public HGCalTowerProcessorBase {
       towerMapsPtrs.emplace_back(towerMapCollHandle, i);
     }
 
-    /* create additional TowerMaps from the unclustered TCs */
+    if (includeTrigCells_) {
+      /* create additional TowerMaps from the unclustered TCs */
 
-    // translate our HGCalClusters into HGCalTriggerCells
-    std::vector<edm::Ptr<l1t::HGCalTriggerCell>> trigCellVec;
-    for (unsigned i = 0; i < unclTCsCollHandle->size(); ++i) {
-      edm::Ptr<l1t::HGCalCluster> ptr(unclTCsCollHandle, i);
-      for (const auto& itTC : ptr->constituents()) {
-        trigCellVec.push_back(itTC.second);
+      // translate our HGCalClusters into HGCalTriggerCells
+      std::vector<edm::Ptr<l1t::HGCalTriggerCell>> trigCellVec;
+      for (unsigned i = 0; i < unclTCsCollHandle->size(); ++i) {
+        edm::Ptr<l1t::HGCalCluster> ptr(unclTCsCollHandle, i);
+        for (const auto& itTC : ptr->constituents()) {
+          trigCellVec.push_back(itTC.second);
+        }
       }
-    }
 
-    // fill the TowerMaps with the HGCalTriggersCells
-    l1t::HGCalTowerMapBxCollection towerMapsFromUnclTCs;
-    towermap2D_->buildTowerMap2D(trigCellVec, towerMapsFromUnclTCs);
+      // fill the TowerMaps with the HGCalTriggersCells
+      l1t::HGCalTowerMapBxCollection towerMapsFromUnclTCs;
+      towermap2D_->buildTowerMap2D(trigCellVec, towerMapsFromUnclTCs);
 
-    /* merge the two sets of TowerMaps */
-    unsigned int towerMapsPtrsSize = towerMapsPtrs.size();
-    for (unsigned int i = 0; i < towerMapsFromUnclTCs.size(); ++i) {
-      towerMapsPtrs.emplace_back(&(towerMapsFromUnclTCs[i]), i + towerMapsPtrsSize);
-    }
+      /* merge the two sets of TowerMaps */
+      unsigned int towerMapsPtrsSize = towerMapsPtrs.size();
+      for (unsigned int i = 0; i < towerMapsFromUnclTCs.size(); ++i) {
+        towerMapsPtrs.emplace_back(&(towerMapsFromUnclTCs[i]), i + towerMapsPtrsSize);
+      }
 
-    /* call to towerMap3D clustering */
-    towermap3D_->buildTowerMap3D(towerMapsPtrs, collTowers);
+      /* call to towerMap3D clustering */
+      towermap3D_->buildTowerMap3D(towerMapsPtrs, collTowers);
+    } else {
+      /* call to towerMap3D clustering */
+      towermap3D_->buildTowerMap3D(towerMapsPtrs, collTowers);
+    }
```

Comment:
This could be simplified by removing the `else` and moving `towermap3D_->buildTowerMap3D(towerMapsPtrs, collTowers);` outside the `if` statement, since it is called in any case.

Reply:
Yes, this is what I had tried first. But this fails with the following error message (in the default case when partial module sums are used and the code enters the `if` part):

`StdException HGCalTowerMap: Trying to add HGCalTowerMaps with different bins: 2592 and 0`

I think this is because `towerMapsFromUnclTCs` only lives within the `if` statement now, and dies outside. What I could do, though, is construct this object outside the `if` statement, which would work too.

Reply:
I see. I think it is fine as it is then.